File: app/src/main/java/org/movieos/feeder/sync/Subscription.java
package org.movieos.feeder.sync;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

import io.realm.RealmObject;
import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

public class Subscription extends RealmObject implements IntegerPrimaryKey {
    @SerializedName("id")
    @PrimaryKey
    int mId;

    @Required
    @SerializedName("created_at")
    Date mCreatedAt;

    @Index
    @SerializedName("feed_id")
    int mFeedId;

    @Required
    @SerializedName("title")
    String mTitle;

    @Required
    @SerializedName("feed_url")
    String mFeedUrl;

    @Required
    @SerializedName("site_url")
    String mSiteUtl;

    public Subscription() {
    }

    public int getId() {
        return mId;
    }

    public Date getCreatedAt() {
        return mCreatedAt;
    }

    public int getFeedId() {
        return mFeedId;
    }

    public String getTitle() {
        return mTitle;
    }

    public String getFeedUrl() {
        return mFeedUrl;
    }

    public String getSiteUtl() {
        return mSiteUtl;
    }
}

File: app/src/main/java/org/movieos/bluereader/model/Subscription.kt
package org.movieos.bluereader.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import java.util.*

@Entity(
        indices = arrayOf(
                Index("feedId")
        )
)
data class Subscription(
        @SerializedName("id")
        @PrimaryKey
        var id: Int,

        @SerializedName("created_at")
        var createdAt: Date?,

        @SerializedName("feed_id")
        var feedId: Int,

        @SerializedName("title")
        var title: String?,

        @SerializedName("feed_url")
        var feedUrl: String?,

        @SerializedName("site_url")
        var siteUrl: String?,

        // This is locally generated, not fetched from the server
        var unreadCount: Int
)
